model/states.go

```diff
@@ -18,8 +18,29 @@ import (
 	"encoding/json"
 	"fmt"
 	"strings"
+
+	validator "github.com/go-playground/validator/v10"
+	val "github.com/serverlessworkflow/sdk-go/v2/validator"
 )
 
+func init() {
+	val.GetValidator().RegisterStructValidation(stateStructLevelValidation, State{})
+}
+
+func stateStructLevelValidation(structLevel validator.StructLevel) {
```

Comment:
I'm assuming we will have more validations in this method since the name is too broad, right? For example, transitioning to a state that doesn't exist. I understand that we can evolve as we go, and not necessarily do everything in this PR.

Reply:
Yes, we can have a `states_validator.go` and think about the naming. It's ok to keep this name now and add a comment here detailing that this function can be broader and maybe be composed of small ones in the future that do the heavy lifting.